node/src/actor_system_factory.rs

```diff
@@ -1605,6 +1616,39 @@ mod tests {
         system.run();
     }
 
+    #[should_panic(expected = "1: Automap failure: prefixProtocolError(\"Booga\")")]
+    #[test]
+    fn handle_automap_error_handles_crashing_errors() {
+        running_test();
+        let logger = Logger::new("handle_automap_error_handles_non_crashing_errors");
+
+        ActorSystemFactoryToolsReal::handle_automap_error(
+            &logger,
+            "prefix",
+            AutomapError::ProtocolError("Booga".to_string()),
+        );
+    }
+
+    #[test]
+    fn handle_automap_error_handles_non_crashing_errors() {
+        init_test_logging();
+        let make_params_arc = Arc::new(Mutex::new(vec![]));
+        let automap_control_factory = AutomapControlFactoryMock::new()
+            .make_params(&make_params_arc)
+            .make_result(Box::new(AutomapControlMock::new()));
+        let mut subject = ActorSystemFactoryToolsReal::new();
+        subject.automap_control_factory = Box::new(automap_control_factory);
+
+        let _ = subject.make_automap_control(None, vec![]);
+
+        let mut make_params = make_params_arc.lock().unwrap();
+        let change_handler = make_params.remove(0).1;
```

Comment:
Sometimes mentioning even the other items in the tuple is valuable, giving some more context to the test. Not sure this is the case, easily not.

Reply:
Okay, but it's not really meaningful in this case, as you'll see...